script.js

```diff
@@ -53,81 +79,247 @@ my_widget_script =
             }
             return true;
           }
-         
+
           $('#the_form tbody tr').each(function() {
-              if (isEmpty(this)) {
-                $(this).remove();
-              }
-            })
+            if (isEmpty(this)) {
+              $(this).remove();
+            }
+          })
           return;
         }
-       
+
         var nFixed = 2;
-        $('#the_form input[name^=amount]').on('keyup change', function() {
+
+        /*
+			Amount Field On Blur Handler
+		*/
+        $('#the_form input[name^=amount]').on('blur', function() {
           var tr = $(this).closest('tr');
-          var equiv = $('input[name^=equivalents]', tr);
-          if (equiv.length != 1)
-            throw ('Found the incorrect number of equiv inputs: ' + equiv.length)
-          equiv = equiv.first();
-          var moles1 = $('#the_form input[name=moles1_formula]');
+          console.log('In amount blur handler'+ tr.attr('class'))
+
+          current_class = tr.attr('class')
```

Comment:
`tr.attr('class')` is clearer and more succinct and almost as fast as `current_class`. It is only used once. Don't bother giving it a name, as it only makes the code harder to read.